### PR TITLE
Made the documentation more idiomatic and fixed some small grammatical errors

### PR DESCRIPTION
**Pull request by w-henderson**
I've found this library extremely useful in several projects and I'm very grateful for it. I have however found that, as a native English speaker, the documentation does not read particularly clearly in parts and there are some very minor grammatical errors. I have gone through all of the documentation and amended the grammar and phrasing in some places to improve the overall quality of the documentation and I hope this is useful to the project.

I found that there was quite a mix of American and British English spellings in the documentation (for example, "initialise" and "initialize" were used interchangeably). More of the spelling used British English so I chose to change the American spellings into the British ones to make the documentation more consistent. If you would like me to change them all to the American spellings that is equally fine so just let me know.

### CODE REVIEW

File: src/os/windows/mod.rs [hunk 1]
```diff
@@ -234,12 +234,12 @@ impl Library {
...
     }
 
-    /// Get a pointer to function or static variable by ordinal number.
+    /// Get a pointer to a function or static variable by ordinal number.
     ///
     /// # Safety
     ///
     /// Users of this API must specify the correct type of the function or variable loaded. Using a
-    /// `Symbol` with a wrong type is undefined.
+    /// `Symbol` with a wrong type causes undefined behaviour.
```
Comment (nagisa) [inline src/os/windows/mod.rs:242]: Using a `Symbol` with a wrong type here cannot cause undefined behaviour in the way "causes undefined behaviour" is understood in the context of the language specification and frequently the standard library documentation (i.e. compiler can do whatever it wishes to)

The intent we'd like to convey here is instead that this library is unable to describe what exactly would happen when a `Symbol` with an invalid type is used. But either way the compiler does not receive a permission to do whatever it wishes to.

For that reason I've avoided including the “behaviour” in this documentation. If the old sentence reads poorly, I would rather we delete it entirely. I don't feel like it adds much value.

File: src/os/windows/mod.rs [hunk 2]
```diff
@@ -178,8 +178,8 @@ impl Library {
     ///
     /// # Safety
     ///
-    /// When a library is loaded initialization routines contained within the library are executed.
-    /// For the purposes of safety, execution of these routines is conceptually the same calling an
+    /// When a library is loaded initialisation routines contained within the library are executed.
```
Comment (nagisa) [inline src/os/windows/mod.rs:181]: Doesn't this need a "the" or a comma somewhere in between "loaded" and "initialization"?

File: src/os/windows/mod.rs (hunk 1 above)
Comment (nagisa) [reply src/os/windows/mod.rs:242]: Lets remove it then. Maybe sometime down the line I'll come up with a clarification that's both easy to understand and conveys the intent accurately.

**Comment by w-henderson** [comment]
No problem, thanks for merging! I'm taking part in Hacktoberfest this year so if you don't mind could you tag this `hacktoberfest-accepted`? If not it's not a problem, I'm pleased to have been able to contribute to such a useful project.

**Comment by nagisa** [comment]
Done!

**Pull request merged**